python/utilities/submit_chain/submit_chain.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+"""Submits a chain to a list of logs."""
+
+import base64
+import hashlib
+import sys
+
+import json
+import logging
+import gflags
+
+from ct.client import log_client
+from ct.crypto import cert
+from ct.crypto import error
+from ct.crypto import pem
+from ct.crypto import verify
+from ct.proto import client_pb2
+from ct.serialization import tls_message
+
+FLAGS = gflags.FLAGS
+
+gflags.DEFINE_string("log_list", None, "List of logs to submit to.")
+gflags.DEFINE_string("chain", None, "Certificate chain to submit (PEM).")
+gflags.DEFINE_string("log_schema", "http", "Log schema (http/https)")
+gflags.DEFINE_string("output", None, "output file for sct_list")
+gflags.MarkFlagAsRequired("log_list")
+gflags.MarkFlagAsRequired("chain")
+gflags.MarkFlagAsRequired("output")
+
+def _read_ct_log_list(log_list_file):
+    """Parses the log list JSON, returns a log url to key map."""
+    try:
+        log_list_json = json.loads(log_list_file)
+        log_url_to_key = {}
+        for log_info in log_list_json['logs']:
+            log_url_to_key[FLAGS.log_schema + '://' + log_info['url']] = (
+                base64.decodestring(log_info['key']))
+        return log_url_to_key
+    except (OSError, IOError) as io_exception:
+        raise Exception('Could not read log list file %s: %s' %
+                        (log_list_file, io_exception))
+
+def _submit_to_single_log(log_url, full_chain):
+    """Submits the chain to a single log specified by log_url."""
+    ct_client = log_client.LogClient(log_url, connection_timeout=10)
+    res = None
+    try:
+        res = ct_client.add_chain(full_chain)
+    except log_client.HTTPError as err:
+        logging.info('Skipping log %s because of error: %s\n', log_url, err)
+    return res
+
+def _map_log_id_to_verifier(log_list):
+    """Returns a map from log id to verifier object from the log_list."""
+    log_id_to_verifier = {}
+    for log_key in log_list.values():
+        key_id = hashlib.sha256(log_key).digest()
+        key_info = client_pb2.KeyInfo()
+        key_info.type = client_pb2.KeyInfo.ECDSA
+        key_info.pem_key = pem.to_pem(log_key, 'PUBLIC KEY')
+        log_id_to_verifier[key_id] = verify.LogVerifier(key_info)
+    return log_id_to_verifier
+
+def _submit_to_all_logs(log_list, certs_chain):
+    """Submits the chain to all logs in log_list and validates SCTs."""
+    log_id_to_verifier = _map_log_id_to_verifier(log_list)
+
+    chain_der = [c.to_der() for c in certs_chain]
+    logging.info("chain der: %s", chain_der)
+    raw_scts_for_cert = []
+    for log_url in log_list.keys():
+        res = _submit_to_single_log(log_url, chain_der)
+        if res:
+            raw_scts_for_cert.append(res)
+        else:
+            logging.info("No SCT from log %s", log_url)
+
+    validated_scts = []
+    for raw_sct in raw_scts_for_cert:
+        key_id = raw_sct.id.key_id
+        try:
+            log_id_to_verifier[key_id].verify_sct(raw_sct, certs_chain[0])
```

Comment:
In line 73 I'm checking if res is not None and only then adding it to raw_scts_for_cert.